runtime/lua/vim/treesitter.lua

```diff
@@ -38,61 +46,52 @@ function Parser:_on_lines(bufnr, changed_tick, start_row, old_stop_row, stop_row
   end
 end
 
+--- Sets the included ranges for the current parser
+--
+-- @param ranges A table of nodes that will be used as the ranges the parser should include.
 function Parser:set_included_ranges(ranges)
   self._parser:set_included_ranges(ranges)
   -- The buffer will need to be parsed again later
   self.valid = false
 end
 
-local M = {
-  parse_query = vim._ts_parse_query,
-}
+-- TODO(vigoux): not that great way to do it, but that __index method bothers me...
+local M = vim.tbl_extend("error", query, lang)
 
 setmetatable(M, {
   __index = function (t, k)
       if k == "TSHighlighter" then
-        t[k] = require'vim.tshighlighter'
+        t[k] = require'vim.treesitter.highlighter'
```

Comment:
@tjdevries does it make sense to keep both at the same time ? as this API was not released yet, I would vote for removing it. The risk is for newcomers to use the old one or that both confuse them.

Reply:
I'm fine w/ removing it, was just thinking we may remove it before the next release and give people a week or two to update w/out breaking. Fine w/ just a message that tells them what to do (and then removing the message shortly after)

Reply:
(we will delete the warning later. Decided offline)